internal/manifest/ready_check.go

```diff
@@ -232,3 +72,31 @@ func isDeploymentReady(clt declarativev2.Client, resources []*resource.Info) boo
 	}
 	return false
 }
+
+func getPodsForDeployment(clt declarativev2.Client, deploy *apiappsv1.Deployment) (*apicorev1.PodList, error) {
+	podList := &apicorev1.PodList{}
+	listOptions := &client.ListOptions{
+		Namespace:     deploy.Namespace,
+		LabelSelector: k8slabels.SelectorFromSet(deploy.Spec.Selector.MatchLabels),
+	}
+	if err := clt.List(context.TODO(), podList, listOptions); err != nil {
```

Comment:
you need to use context from `Run(ctx context.Context, ...`

Reply:
Good catch!